components/renku_data_services/project/orm.py

```diff
@@ -79,3 +73,35 @@ class ProjectRepositoryORM(BaseORM):
         ForeignKey("projects.id", ondelete="CASCADE"), default=None, index=True
     )
     project: Mapped[Optional[ProjectORM]] = relationship(back_populates="repositories", default=None)
+
+
+class ProjectSlug(BaseORM):
+    """Project and namespace slugs."""
+
+    __tablename__ = "project_slugs"
+    __table_args__ = (Index("project_slugs_unique_slugs", "ltst_ns_slug_id", "slug", unique=True),)
```

Comment:
I think we are missing code that updates ltst_ns_slug when a namespace is renamed. so this would allow to violate this unique constaint. At least I couldn't find a place where we update this